api/src/api/users/user_routes.py

```diff
@@ -231,3 +234,36 @@ def user_get_saved_opportunities(db_session: db.Session, user_id: UUID) -> respo
     saved_opportunities = get_saved_opportunities(db_session, user_id)
 
     return response.ApiResponse(message="Success", data=saved_opportunities)
+
+
+@user_blueprint.post("/<uuid:user_id>/saved-searches")
+@user_blueprint.input(UserSaveSearchRequestSchema, location="json")
+@user_blueprint.output(UserSaveSearchResponseSchema)
+@user_blueprint.doc(responses=[200, 401])
+@user_blueprint.auth_required(api_jwt_auth)
+@flask_db.with_db_session()
+def user_save_search(
+    db_session: db.Session, user_id: UUID, json_data: dict
+) -> response.ApiResponse:
+    logger.info("POST /v1/users/:user_id/saved-searches")
+
+    user_token_session: UserTokenSession = api_jwt_auth.current_user  # type: ignore
+
+    # Verify the authenticated user matches the requested user_id
+    if user_token_session.user_id != user_id:
+        raise_flask_error(401, "Unauthorized user")
+
+    saved_search = create_saved_search(db_session, user_id, json_data)
+
+    with db_session.begin():
+        db_session.add(saved_search)
```

Comment:
Should be:
```suggestion
    with db_session.begin():
        create_saved_search(db_session, user_id, json_data)
```

Except for beginning the session, don't want any DB logic in this code